upup/pkg/fi/nodeup/nodetasks/pull_image.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodetasks
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"k8s.io/klog/v2"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/local"
+)
+
+// PullImageTask is responsible for pulling a docker image
+type PullImageTask struct {
+	Name    string
+	Runtime string
+}
+
+var _ fi.Task = &PullImageTask{}
+var _ fi.HasDependencies = &PullImageTask{}
+
+func (t *PullImageTask) GetDependencies(tasks map[string]fi.Task) []fi.Task {
+	// LoadImageTask depends on the docker service to ensure we
+	// sideload images after docker is completely updated and
+	// configured.
+	var deps []fi.Task
+	for _, v := range tasks {
+		if svc, ok := v.(*Service); ok && svc.Name == containerdService {
+			deps = append(deps, v)
+		}
+		if svc, ok := v.(*Service); ok && svc.Name == dockerService {
+			deps = append(deps, v)
+		}
+	}
+	return deps
+}
+
+func (e *PullImageTask) Find(c *fi.Context) (*PullImageTask, error) {
+	klog.Warningf("LoadImageTask checking if image present not yet implemented")
```

Comment:
This isn't information that would be important when diagnosing a failure, so warning level is too high.